vendors/vendor/boards/board/ports/ota_pal_for_aws/ota_pal.h

```diff
@@ -0,0 +1,229 @@
+/*
+ * FreeRTOS V202012.00
+ * Copyright (C) 2019 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+/**
+ * @file  ota_pal.h
+ * @brief Function declarations for the functions in ota_pal.c
+ */
+
+#ifndef OTA_PAL_H_
+#define OTA_PAL_H_
+
+#include "ota.h"
+
+/**
+ * @brief Abort an OTA transfer.
+ *
+ * Aborts access to an existing open file represented by the OTA file context pFileContext. This is
```

Comment:
This information is nice but much if it is needed near the FIX ME (prefer TODO) so the user knows what the function is to be when finished.

Reply:
To be clear, are you recommending that I copy the function headers in the ota_pal.c file?

Reply:
I don't think the entire function header needs to go there.  But I see some important information like what pFileContext is, what the allowed return codes are.  I also think the purpose of the function should be explained.  Perhaps that is in the brief.

Reply:
Is this required or do you think it would be apparent that the information is in the header file? We also have a porting guide that we expect people to follow for implementing a port of the OTA library. My hesitation is that I want to avoid having two places that are describing the same thing for maintenance reasons.

Reply:
Ok